### ISSUE TITLE
[SANDBOX PROJECT ONBOARDING] HAMi

### ISSUE THREAD

**Issue by Cmierly** [cncf/sandbox#132]
# Welcome to CNCF Project Onboarding!

This is an issue created to help onboard your project into the CNCF after the TOC has voted to accept your project into the Sandbox.

We would like your project to complete onboarding within one month of acceptance.

Please track your progress by using "Quote reply" to create your own copy of this checklist in an issue, so that you can update the status as you finish items.

## Review and understand

- [x] The [Technical Leadership Principles](https://github.com/cncf/toc/blob/main/PRINCIPLES.md#technical-leadership-principles) that outlines the expected behavior for any maintainer in a leadership role.
- [x] The [project proposal process and requirements](https://github.com/cncf/toc/blob/main/process/README.md).
- [x] The [services available for your project at the CNCF](https://contribute.cncf.io/resources/project-services/).
- [x] The [CNCF IP Policy](https://github.com/cncf/foundation/blob/main/charter.md#11-ip-policy).
- [x] The [trademark guidelines](https://www.linuxfoundation.org/legal/trademark-usage).
- [x] The [license allowlist](https://github.com/cncf/foundation/blob/main/allowed-third-party-license-policy.md#approved-licenses-for-allowlist).
- [x] The [online program guidelines](https://github.com/cncf/foundation/blob/main/online-programs-guidelines.md).
- [x] Optional: [Book time with CNCF staff](http://project-meetings.cncf.io) for any onboarding questions.

## Contribute and transfer

- [x] Move your project to its own separate neutral GitHub organization. This will make it transferable to the CNCF's GitHub Enterprise account. If it's already in a GHE account, you will need to remove it from that first.
- [x] Accept the invite to join the CNCF GitHub Enterprise account. We'll then add `thelinuxfoundation` as an organization owner to ensure neutral hosting of your project.
- [x] Migrate your Slack channels (if any) to the [Kubernetes or CNCF Slack workspace](https://slack.com/help/articles/217872578-Import-data-from-one-Slack-workspace-to-another). CNCF staff can help.
- [x] Submit a [pull request](https://github.com/cncf/landscape/pulls) to add your project as a Sandbox project to the [Cloud Native Landscape](https://landscape.cncf.io) by updating [landscape.yml](https://github.com/cncf/landscape/blob/master/landscape.yml) following these [instructions](https://github.com/cncf/landscape2/blob/main/docs/config/data.yml).
- [x] Transfer [your domain to the CNCF](https://jira.linuxfoundation.org/plugins/servlet/theme/portal/2/create/63). The "LF Stakeholder email" is <projects@cncf.io>.
- [x] Transfer any [trademark and logo assets to the Linux Foundation](https://github.com/cncf/foundation/tree/main/agreements) via the Contribution Agreement.
- [x] Submit a [pull request](https://github.com/cncf/artwork) with your artwork.
- [x] Transfer website analytics to <projects@cncf.io>. CNCF staff can help.

## Update and document

- [x] Ensure that [DCO](https://github.com/apps/dco) (preferred) or [CLA](https://github.com/cncf/cla) are enabled for all GitHub repositories of the project.
- [x] Ensure that that the [CNCF Code of Conduct](https://github.com/cncf/foundation/blob/main/code-of-conduct.md) (or your adopted version of it) are explicitly referenced at the project's `README` on GitHub.
- [x] Ensure LF footer is on your website and [guidelines](https://github.com/cncf/foundation/blob/main/website-guidelines.md) are followed (if your project doesn't have a dedicated website, please adopt those guidelines for the `README` file).
- [x] Create a maintainer list and add it to the [aggregated CNCF maintainer list](https://maintainers.cncf.io) via pull request.
- [x] Provide emails for the maintainers to get access to the maintainers mailing list and Service Desk. Email them to <project-onboarding@cncf.io>.
- [x] Start working on [written, open governance](https://contribute.cncf.io/maintainers/governance/).
- [x] Start on an [OpenSSF Best Practices Badge](https://www.bestpractices.dev/en).

## CNCF staff tasks

- [x] Add the project to [DevStats](https://devstats.cncf.io/).
- [x] Add the project to [CLOmonitor](https://clomonitor.io/).
- [x] Add the project to [LFX Insights](https://insights.lfx.linuxfoundation.org/). This is done by adding a read-only app to your GitHub organization once it's in CNCF GHE.
- [x] Add the project to [LFX Project Control Center](https://projectadmin.lfx.linuxfoundation.org/project/a0941000002wBz4AAE).
- [x] Add a license scanning tool, like [FOSSA](https://fossa.com/) or [Snyk](https://snyk.io/).
- [x] Invite developers to the [#maintainers-circle](https://app.slack.com/client/T08PSQ7BQ/C014YQ8CDCG) Slack channel.
- [x] Send a welcome email to confirm maintainer list access.

**Comment by wawa0210** [comment]
@RobertKielty 

How to transfer Slack channels? Can you help me? Thanks

Also, I didn’t receive the email from FOSSA xiaozhang0210@hotmail.com

**Comment by wawa0210** [comment]
friendly ping @RobertKielty @idvoretskyi @cynthia-sg 

The following tasks are left to do
* Migrate your Slack channels (if any) to the [Kubernetes or CNCF Slack workspace]
* Add the project to LFX Project Control Center.
* Add a license scanning tool, like FOSSA or Snyk.

**Comment by RobertKielty** [comment]
I have just sent out invites for team to both FOSSA and Snyk. Let me know when the invites are accepted.

**Comment by wawa0210** [comment]
> I have just sent out invites for team to both FOSSA and Snyk. Let me know when the invites are accepted.

I received an invitation email from Snyk, but not from FOSSA

**Comment by wawa0210** [comment]
> [@wawa0210](https://github.com/wawa0210) for the Slack channels, please file a ticket to the CNCF ServiceDesk, we'll handle it there.
> 

![Image](https://github.com/user-attachments/assets/1d2e7b93-afa0-4045-819c-77c0122d2c26)

I doesn't have access to Jira on cncfservicedesk.atlassian.net. Need help

**Comment by Cmierly** [comment]
> > [@wawa0210](https://github.com/wawa0210) for the Slack channels, please file a ticket to the CNCF ServiceDesk, we'll handle it there.
> 
> ![Image](https://github.com/user-attachments/assets/1d2e7b93-afa0-4045-819c-77c0122d2c26)
> 
> I doesn't have access to Jira on cncfservicedesk.atlassian.net. Need help

I'll look into this today!

**Comment by RobertKielty** [comment]
[@wawa0210](https://github.com/wawa0210) I have sent you a direct message on Cloud Native Slack (using wawa0219) feel free to repond to me there if you like.

I would like to review the work already done on getting license scans done.

I can see we have the main repo added to the HAMi team in Snyk. (there are two License issues reported that will need to be reviewed and addressed) [Project HAMi license scan results](https://app.snyk.io/org/hami/reporting?v=1&context[page]=issues-detail&issue_status=%255B%2522Open%2522%255D&issue_by=Severity&table_issues_detail_cols=SEVERITY%257CSCORE%257CCVE%257CCWE%257CPROJECT%257CEXPLOIT%2520MATURITY%257CCOMPUTED%2520FIXABILITY%257CINTRODUCED%257CSNYK%2520PRODUCT&issue_type=%255B%2522License%2522%255D)

There are 12 repos over on https://github.com/orgs/Project-HAMi/repositories?type=all 

All 12 repos will need to be scanned. 

Using Snyk to scan all the repos in the Project-HAMi GH org would allow us to mark this task complete. 

But I am happy to get you setup on FOSSA if you prefer.

**Comment by RobertKielty** [comment]
For Slack migration, I have signed up to the current HAMi Slack to review the channels there.

We have a small number of candidate channels to migrate with a message history dating back to April 8th this year.

We have 

- All (全体) 
- Random (全体) 
- opensource
- dev

We should look at migrating opensource and dev accross. Suggested steps 

On HAMi Slack 
1. rename dev and opensource channels tp HAMi-dev and HAMi-opensource
2. Get a Workspace Owner to export HAMi-dev and HAMi-opensource; channel message history and membership. In order to get a full export that include the users the HAMi Slack account MUST BE an Owner account. An Admin account will not work for this.

Then reach out to me and share the export file with me. I will then attempt to import it to the CNCF Slack.

**Comment by wawa0210** [comment]
> For Slack migration, I have signed up to the current HAMi Slack to review the channels there.
> 
> We have a small number of candidate channels to migrate with a message history dating back to April 8th this year.
> 
> We have
> 
> * All (全体)
> * Random (全体)
> * opensource
> * dev
> 
> We should look at migrating opensource and dev accross. Suggested steps
> 
> On HAMi Slack
> 
> 1. rename dev and opensource channels tp HAMi-dev and HAMi-opensource
> 2. Get a Workspace Owner to export HAMi-dev and HAMi-opensource; channel message history and membership. In order to get a full export that include the users the HAMi Slack account MUST BE an Owner account. An Admin account will not work for this.
> 
> Then reach out to me and share the export file with me. I will then attempt to import it to the CNCF Slack.

Thank you for your very patient guidance. The slack export data has been sent to you via slack.

**Comment by wawa0210** [comment]
> FOSSA - all proects imported Slack - two channels imported we enoucntered the usual User importation issues to instead we imported the message history without mergin the users so that the migration happened. [@wawa0210](https://github.com/wawa0210) will announce the swicth over to the cloud naitve worspace in the HAMi workspace.

Very nice cooperation, very patient. We also work together as a team. I found that I urgently need to improve my English speaking skills 😄

**Comment by RobertKielty** [comment]
@wawa0210 yes it was great to collaborate with you today! You're English is way better thank my Mandarin, no worries there! 😆 

I should have LFX Insights app intructions for you tomorrow, and I've shared the CI/CD resourcing questions with the colleagues on th Project Team, I should have guidance for you tomorrow on that.

**Comment by Cmierly** [comment]
> [@Cmierly](https://github.com/Cmierly) help me, i need some support
> 
> For this task: Transfer any [trademark and logo assets to the Linux Foundation](https://github.com/cncf/foundation/tree/main/agreements).
> 
> Question: We haven't registered any trademarks yet. so i guess this task is passed naturally?
> 
> And for this task: Transfer website analytics to [projects@cncf.io](mailto:projects@cncf.io) Question: Our website is run on a VPS(IP: 193.203.215.25) by using command 'npm run serve'. so it probably doesn't have a website analytics. how can i pass this task?

Hi!
Sorry about the late reply but you will need to fill out the "no reg trademark" agreement in order for this task to be marked off.

**Comment by wawa0210** [comment]
> fill out the "no reg trademark" agreement

Could you please guide us on how we should proceed?

**Comment by Cmierly** [comment]
@wawa0210 We are working on a new process for receiving trademark agreements. I will update you once we have more information.
I apologize for the delay!

**Comment by RobertKielty** [comment]
@wawa0210 I have logged a request to have HAMi added to LFX Insights, thank you for your patience on this!

cc @Cmierly @krook

**Comment by wawa0210** [comment]
friendly ping

I would like to ask about the current HAMi sandbox onboarding progress. All tasks should be completed now.

@Cmierly @krook

**Comment by wawa0210** [comment]
> Hi [@wawa0210](https://github.com/wawa0210) yes, you've completed the tasks on your side, thank you!
> 
> But, we have a set of serial tasks to complete on our side with your help on one of them.
> 
> We aim to resolve these by the end of the month:
> 
> * [ ]  Adding the project to [LFX Insights](https://insights.lfx.linuxfoundation.org/). This is done by adding a read-only app to your GitHub organization once it's in CNCF GHE.
> 
> Which is dependent on us:
> 
> * [ ]  Adding the project to [LFX Project Control Center](https://projectadmin.lfx.linuxfoundation.org/project/a0941000002wBz4AAE)
> 
> Which is in turn dependent on both of us completing the new documents to:
> 
> * [ ]  Transfer any [trademark and logo assets to the Linux Foundation](https://github.com/cncf/foundation/tree/main/agreements).
> 
> The blocker here is on our side to get that trademark part done and sent over to you because it's a new doc that's being finalized.

Hahahaha, I got it. Once the trademark part is completed, everything can be moved forward.

**Comment by krook** [comment]
Sorry for the delay @wawa0210. The new 2025 "Project Contribution Agreement" is now available. 

This replaces the trademark agreements mentioned in the original checklist. 

I'll be sending an email to the contacts listed in the Sandbox application issue to handle next steps. 

Please review the instructions in the email, fill out the Google Doc, and then we can it them completed via Docusign. 

Once done, that will unblock us from continuing the rest of these Sandbox onboarding tasks. 

We can address any questions about the Agreement over email.

**Comment by krook** [comment]
All onboarding tasks are now complete 🎉

**Comment by riaankleinhans** [comment]
- [x] Add project's groups.io cncf-sandbox-projects list to [cncf-sandbox-projects@cncf.io](https://groups.google.com/a/cncf.io/g/cncf-sandbox-projects)
